src/features/orders/orderApi.ts

```diff
@@ -118,9 +120,26 @@ export async function takeOrder(
 }
 
 export function orderSortingFunction(a: Order, b: Order) {
+  const now = Date.now();
+  const aTimeToExpiry = now - parseInt(a.expiry);
+  const bTimeToExpiry = now - parseInt(b.expiry);
+
+  // If any of the orders come in with an expiry that is less than our required
+  // buffer, put them to the bottom of the list.
+  if (
+    aTimeToExpiry < RFQ_EXPIRY_BUFFER_MS &&
+    bTimeToExpiry >= RFQ_EXPIRY_BUFFER_MS
+  ) {
+    return 1; // prefer B
+  } else if (
+    bTimeToExpiry < RFQ_EXPIRY_BUFFER_MS &&
+    aTimeToExpiry >= RFQ_EXPIRY_BUFFER_MS
+  ) {
+    return -1; // prefer A
+  }
   // If tokens transferred are the same
   if (a.signerAmount === b.signerAmount && a.senderAmount === b.senderAmount) {
-    return parseInt(b.expiry) - parseInt(a.expiry);
+    return aTimeToExpiry - bTimeToExpiry;
```

Comment:
This fixed the following ordersSlice test:

```
should select orders based on longest expiry if token amounts are equal
```

Reply:
Ah, it's making the tests pass but not for the right reasons. We pass strings for expiries in the tests so this is definitely needed.